src/pages/home/report/ReportActionsView.tsx

```diff
@@ -137,20 +137,24 @@ function ReportActionsView({
 
     // Get a sorted array of reportActions for both the current report and the transaction thread report associated with this report (if there is one)
     // so that we display transaction-level and report-level report actions in order in the one-transaction view
-    const combinedReportActions = useMemo(() => {
+    const [combinedReportActions, parentReportActionForTransactionThread] = useMemo(() => {
         if (isEmptyObject(transactionThreadReportActions)) {
-            return allReportActions;
+            return [allReportActions, null];
         }
 
         // Filter out the created action from the transaction thread report actions, since we already have the parent report's created action in `reportActions`
         const filteredTransactionThreadReportActions = transactionThreadReportActions?.filter((action) => action.actionName !== CONST.REPORT.ACTIONS.TYPE.CREATED);
+        const moneyRequestAction = allReportActions.find((action) => {
+            const actionType = (action as OnyxTypes.OriginalMessageIOU).originalMessage?.type ?? '';
+            return actionType === CONST.IOU.REPORT_ACTION_TYPE.CREATE || actionType === CONST.IOU.REPORT_ACTION_TYPE.TRACK || ReportActionsUtils.isSentMoneyReportAction(action);
+        });
```

Comment:
@ishpaul777 The reason because the filter here was wrong. Before it was `!ReportActionsUtils.isSentMoneyReportAction(action);`. Update this condition correctly, please help to check again.

Reply:
@ishpaul777 For combine report, the only `moneyRequestAction` is the parentAction of transaction thread report. The check `return actionType === CONST.IOU.REPORT_ACTION_TYPE.CREATE || actionType === CONST.IOU.REPORT_ACTION_TYPE.TRACK || ReportActionsUtils.isSentMoneyReportAction(action);` is used to find that. 

It this the negation of the check that we used to filter out this action type below. We also can use the filter `action.reportActionID === transsactionThreadReport.parentReportActionID` here the result is the same.

I don't like the solution to use `ReportActionsUtils.getAllReportActions(report.reportID)` because we don't subscribe `allReportActions` from Onyx in `ReportActionItem`.

Reply:
> this doesn't seem true, as you see in video reportActionID is different than passed as prop, also the issue is still reproducible



@ishpaul777 Can you share the full step for this case? I can't reproduce it in this PR now.


> This can also work but i prefer we avoid drilling prop when we can

we already have `allReportActions` in `ReportActionsView` so I think we should get on this and because we can have many `ReportActionItem`, subscribing again `allReportActions` in each `ReportActionItem` isn't good for perf.

Reply:
@ishpaul777 Any update here.

Reply:
> can you please first share a screen recording of the bug that exists now


As we can see in this condition, we will show the total in combine report if the `transactionCurrency` is different from the currency of the workspace.
https://github.com/Expensify/App/blob/ed7029b33c5711736569aea8981e21fc68e12d89/src/pages/home/report/ReportActionItem.tsx#L760


But on staging, we can see in both case the total always appears. The problem is what I mentioned in my proposal, `transaction` is always `undefined` in `ReportActionItem`.

cc @pecanoro 


<img width="1512" alt="Screenshot 2024-04-16 at 17 25 52" src="https://github.com/Expensify/App/assets/161821005/b6b136fb-9340-4f16-bfd8-fa69e1aa4b3d">


<img width="1512" alt="Screenshot 2024-04-16 at 17 24 41" src="https://github.com/Expensify/App/assets/161821005/dacc7fe4-c830-4d67-adf6-dd82204ae519">


> okay i gave this a thought lets move with passing the reportaction filtered with action.reportActionID === transsactionThreadReport.parentReportActionID

I updated it.